src/identity-utils.ts

```diff
@@ -17,74 +30,101 @@ export type IParseCachedIdentityResponse = (
     identityMethod: string,
     knownIdentities: IKnownIdentities,
     fromCachedIdentity: boolean
-) => void
+) => void;
 
 export interface IKnownIdentities extends UserIdentities {
     device_application_stamp?: string;
 }
 
 export interface ICachedIdentityCall {
+    // https://go.mparticle.com/work/SQDSDKS-6672
     responseText: string;
     status: number;
     expireTimestamp: number;
 }
 
+// https://go.mparticle.com/work/SQDSDKS-6568
+// Temporary adapter to convert the XMLHttpRequest response to the IIdentityResponse interface
+export const xhrIdentityResponseAdapter = (
+    possiblyXhr: XMLHttpRequest | IIdentityResponse
+): IIdentityResponse => {
+    // This might be an actual xhr object, so we need to parse it
+    if (!possiblyXhr.hasOwnProperty('expireTimestamp')) {
```

Comment:
suggestion: when i see a `if !() return`, i consider flipping it. this looks like a good candidate, since the else is a 1 line return

something like ```
if (hasProp expireTimestamp) return possiblyXhr 
return {
   ...
}
```